src/gleam/bit_array.gleam

```diff
@@ -193,3 +193,48 @@ fn do_inspect(input: BitArray, accumulator: String) -> String {
     _ -> accumulator
   }
 }
+
+/// Compare two bit arrays as sequences of bytes.
+///
+/// ## Examples
+///
+/// ```gleam
+/// compare(<<1>>, <<2>>)
+/// // -> Lt
+///
+/// compare(<<"AB":utf8>>, <<"AA":utf8>>)
+/// // -> Gt
+///
+/// compare(<<1, 2:size(2)>>, with: <<1, 2:size(2)>>)
+/// // -> Eq
+/// ```
+///
+/// Only supported on Erlang target for now.
+///
+pub fn compare(a: BitArray, with b: BitArray) -> order.Order {
+  do_compare(a, b)
+}
+
+@external(javascript, "../gleam_stdlib.mjs", "bit_array_compare")
+fn do_compare(a: BitArray, with b: BitArray) -> order.Order {
+  case a, b {
+    <<first_byte, first_rest:bits>>, <<second_byte, second_rest:bits>> ->
+      case int.compare(first_byte, second_byte) {
+        order.Eq -> compare(first_rest, second_rest)
+        result -> result
+      }
+
+    <<>>, <<>> -> order.Eq
+    // First has more items, example: "AB" > "A":
+    _, <<>> -> order.Gt
+    // Second has more items, example: "A" < "AB":
+    <<>>, _ -> order.Lt
+    // This happens when there's unusually sized elements.
+    // Handle these special cases via custom erlang function.
+    first, second ->
+      int.compare(bit_array_to_int(first), bit_array_to_int(second))
```

Comment:
Why is this done in Erlang and not Gleam?

Reply:
I am still learning Gleam, maybe I am missing how can I port 

```erl
bit_array_to_int(A) ->
    Size = bit_size(A),
    <<A1:Size>> = A,
    A1.
```

to Gleam? I tried many different patterns and none worked for me. And since I know Erlang for 10 years longer than Gleam I fallbacked to it 😆